docs/core/diagnostics/snippets/Microsoft.Diagnostics.NETCore.Client/csharp/TraceProcessForDuration.cs

```diff
@@ -0,0 +1,34 @@
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.NETCore.Client;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Threading.Tasks;
+
+public partial class Tracer
+{
+    public void TraceProcessForDuration(int processId, int duration, string traceName)
+    {
+        var cpuProviders = new List<EventPipeProvider>()
+        {
+            new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Informational, (long)ClrTraceEventParser.Keywords.Default),
+            new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", EventLevel.Informational, (long)ClrTraceEventParser.Keywords.None)
+        };
+        var client = new DiagnosticsClient(processId);
+        using (var traceSession = client.StartEventPipeSession(cpuProviders))
+        {
+            Task copyTask = Task.Run(async () =>
+            {
+                using (FileStream fs = new FileStream(traceName, FileMode.Create, FileAccess.Write))
+                {
+                    await traceSession.EventStream.CopyToAsync(fs);
+                }
+            });
+
+            Task.WhenAny(copyTask, Task.Delay(duration * 1000));
```

Comment:
This needs an await, otherwise it's going to exit almost immediately. This method needs to be async at that point. Then you need to check which task returned. If it was copy, we ended the session somehow. Otherwise, we need to give time for the copy to finish by awaiting that one (probably with a timeout) after stopping the session, Otherwise the session will get disposed and the stream will get closed and then the async copy will give you corrupted traces.

The file should probably be created outside the session's using statement to handle failure cases.